### DISCUSSION TITLE
Welcome to d.rymcg.tech Discussions!

### DISCUSSION THREAD

**Discussion by EnigmaCurry** [EnigmaCurry/d.rymcg.tech#25] [Announcements]

**Comment by mcmikemn** [comment]
What's the overarching goals of d.rymcg.tech?

**Reply by EnigmaCurry** [reply]
d.rymcg.tech should be:

 * An easy to use self-hosting platform based upon `docker compose`.
 * A collection of open-source software/servers in docker container form, ready to install.
 * A configuration framework for `docker compose` projects, extensible and usable for
   your own external projects.
 * A common pattern to base new projects upon, for example:

   * *All* config goes in .env
   * All env vars all start with the project name.
   * Each project has a Makefile, with the same common targets: `config`, `install`, etc.
   * Config containers for non-environment configuration.
   * Traefik is already assumed to be installed, it doesn't need to be installed again, etc.

 * Provide well tested default configurations. Streamlined
   configuration wizards should only ask the most important questions,
   leaving the config file editing available for more exhaustive configration.
 * Provide idiomatic command line tools to easily configure and manage
   projects.
 * Reduce the time and effort to create new docker compose projects,
   by offering templates, and making it easy to integrate d.rymcg.tech
   tooling.

Goals for the future:

 * Tag and release stable versions of d.rymcg.tech.
 * Test suite for maintaining projects, preferably automated somehow.
   It is difficult to develop d.rymcg.tech itself and to maintain all
   these projects compatibility at the same time. Things break, thats
   expected, but the problem now is that we don't automatically know
   when that happens.
 * Streamlined howto guides and videos. These guides should focus less
   on the breadth of configuration or options, and provide
   opinionated, sequential, instructions for someone to follow.
